staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -963,6 +964,7 @@ func (r *Request) Do(ctx context.Context) Result {
 	if err != nil {
 		return Result{err: err}
 	}
+	metrics.ResponseSize.Observe(ctx, r.verb, r.URL().Host, float64(len(result.body)))
```

Comment:
does this do the right thing if result is an error result? (result.err != nil)

Reply:
I had the same doubt for latency, and asking people with more experience they told me that latency should take into account errors too.
However, for size, I can't see why do we want to store a zero on a failed request, we didn't have a result so I don't think we should measure a length of nothing

Reply:
result.body is nil on failure, hence I don't think we should measure it,

Reply:
I confirmed there is no null value in prometheus , so this is correct, only measure if it exists

Reply:
it still seems like this should be guarded on something like `if result.err == nil` of `if result.body != nil`

Reply:
ack